### DISCUSSION TITLE
Supported Device?

### DISCUSSION THREAD

**Discussion by UberSlackr** [collin80/SavvyCAN#376] [General]
I'm starting my first project with CAN, and I'm wondering if SavvyCAN will work with a RaspberryPi Hat, such as this one from [Waveshare](https://www.waveshare.com/2-ch-can-hat.htm)

I have can-utils installed and running, but I was looking for more of a GUI based option instead of CLI.. Hoping SavvyCAN will work.
The Pi4 is running Raspbian right now. But if I need to install a different distro for SavvyCAN to work with the Pi HAT I have, then I'll for sure do whatever is needed. 

Thanks for any help given, its greatly appreciated!

**Comment by collin80** [comment]
The raspberry pi, being a linux computer, uses socketcan for the CAN interface. And, SavvyCAN supports socketcan interfaces. So, any socketcan device should work. Currently there is no pre-compiled binary for the RPi but I'm aware that people have been compiling SavvyCAN on RPi4 units so it should work.

**Comment by mickeyl** [comment]
I can confirm it works fine.

**Comment by rapbando** [comment]
Can I see data with a Macchina A0 or only the M2 works?

**Reply by collin80** [reply]
The A0 is supported in SavvyCAN. You need the ESP32RET firmware (also found in my repos). In fact, it works wirelessly. I tend to like the A0 because you can plug it into the OBDII port then connect wirelessly and it's pretty simple and easy that way.

**Comment by bfpa40** [comment]
I just received my RaspberryPi 5 and wanted to use it with SavvyCan however there seems to be some sort of a non-compatibility issue if im reading things right. I currently have a PicoCAN board i wanted to use in the interim while I am waiting on my CANable 2.0 board to arrive. Any insight/assistance would be greatly appreciated!

**Comment by ilomon10** [comment]
Is the MKR CANable 2.0 work with savvycan?
